### ISSUE TITLE
Control + Backspace does not delete the word

### ISSUE THREAD

**Issue by prattay56** [felixse/FluentTerminal#946]
Ctrl-Backspace will not delete the entire word to the left of your cursor in one keystroke.

**Comment by altimmons** [comment]
I suspect this works in powershell, I doubt it works in command, and is unlikely to work in Linux/WSL
you can try typing this:  `TERM=xterm` to the environment, and see if it helps.

I believe this is a funtion that must be implemented by the terminal you are connecting to and not this software.

Correct me if I am wrong.  I do personally love that shortcut, and other similar ones.  But because of that, I have looked into this in the past.

**Comment by prattay56** [comment]
Hello Andy,

It works with Windows powershell
both legacy and new tabbed version.


Regards
Aman Das

**Comment by altimmons** [comment]
Yes - that’s what I said.  It’s environment dependent- not fluent dependent

**Comment by prattay56** [comment]
Hello Andy,

My environment is Windows 10.
In this Fluent Terminal 2nd in the Apps category - below SS.
This doesnot work.
However other terminal in windows as mentioned, work. I was expecting the
same with Fluent Terminal in windows 10 as well.
Let me know if you or the team plans to include this binding in the future
if not considered as a Bug.
[image: image.png]

Aman Das

**Comment by prattay56** [comment]
![image](https://user-images.githubusercontent.com/74849664/129426435-fc31da28-2379-420b-a1b4-6f2d0c698a8f.png)

**Comment by prattay56** [comment]
![rec](https://user-images.githubusercontent.com/74849664/129426780-b8e1f9c8-4e86-4536-bb23-77d59ee43e5a.gif)

**Comment by prattay56** [comment]
Hello Team, wanted to follow up this. 
Let me know if you were able to repro this.

**Comment by prattay56** [comment]
Hello Team, wanted to follow up this.
Any update?

**Comment by hanskokx** [comment]
This is an xterm.js issue.  Go ahead and give it a shot on their demo implementation on their homepage. https://xtermjs.org/

**Comment by jzymx50** [comment]
Using Fluent Terminal today and came across the same issue. 
As an alternative, you can set jumping-between-words shortcut for the PowerShell, and use `control + shift + leftarrow` to select words.

Install PSReadline
```
$ Install-Module -Name PSReadLine -AllowPrerelease
```
Open your powershell profile:
```
$ notepad $PROFILE
```
Add these lines:
```
Import-Module PSReadLine
Set-PSReadLineKeyHandler -Key "Ctrl+RightArrow" -Function ForwardWord
Set-PSReadLineKeyHandler -Key "Ctrl+LeftArrow" -Function BackwardWord
```
Save and close. Now you are ready to go!

**Comment by lwbt** [comment]
For this specific issue I started to use `ctrl+right-arrow` and delete the word with `alt+d`, this also works on PowerShell with PSReadLine. PSReadLine also offers the same functionality as `ctrl+backspace` with `ctrl+w`.

Could someone do a bit of cleanup on the open issues?

- oldest: https://github.com/felixse/FluentTerminal/issues/14
- also about PSReadLine: https://github.com/felixse/FluentTerminal/issues/126

It looks like this is going nowhere without an upstream bug report that can be tracked, if the issue really exist upstream. But I have doubt that it does, comments by some here referring to "upstream" without a linked issue is a non-constructive blame game to me.

To me it looks more like an implementation issue of the component after reading through this thread: https://github.com/xtermjs/xterm.js/issues/1179
